bounded_sum_known_n/BoundedSumKnownN.tex

```diff
@@ -0,0 +1,394 @@
+        \item \texttt{IntDistance} must have trait \texttt{InfCast(T)}. (Note that this bullet point is not needed in this proof, but it is needed in the code so a hint can be constructed; otherwise a binary search would be needed to construct the hint.)
+        \item Variables \texttt{U} and \texttt{L} must be of type \texttt{T}, and we must have $\texttt{L} \leq \texttt{U}$.
+
+\end{itemize}
+\end{itemize}
+
+\subsubsection*{Postconditions}
+\begin{itemize}
+    \item Either a valid \texttt{Transformation} is returned or an error is returned.
+\end{itemize}
+
+\begin{lstlisting}[language=Python, escapechar=|]
+def MakeBoundedSumN(L: T, U: T, n: usize):
+    input_domain = SizedDomain(VectorDomain(IntervalDomain(L, U)), n)
+    output_domain = AllDomain(T)
+    input_metric = SymmetricDistance()|\label{code:symm-dist}|
+    output_metric = AbsoluteDistance(T)|\label{code:abs-dist}|
+    
+    n_ = exact_int_cast(n, T)|\label{code:exact-int-cast}|
+    if checked_mul(L, n_).is_none or checked_mul(U, n_).is_none: 
+        raise Exception('Potential overflow') |\label{line:max}|
+    
+    def relation(d_in: u32, d_out: T) -> bool: |\label{line:rel}|
+        if checked_mul(2, d_out).is_none or checked_mul(d_in, U-L).is_none
+            or checked_sub(U, L).is_none:
+            raise Exception('Overflow occurs in the stability relation')
+        return 2*d_out >= inf_cast(d_in, T) * (U - L)
+    
+    def function(data: Vec[T]) -> T: |\label{line:fn}|
+        return sum(data) |\label{line:sum}|
+    
+    return Transformation(input_domain, output_domain, function, input_metric, output_metric, stability_relation = relation)
+\end{lstlisting}
+
+\section{Proof}
+\subsection{Symmetric Distance}
+\begin{theorem}
+    For every setting of the input parameters \texttt{(L, U, n)} to \texttt{MakeBoundedSumN} such that the given preconditions hold, \texttt{MakeBoundedSumN} raises an exception (at compile time or run time) or returns a valid transformation with the following properties:
+    \begin{enumerate}
+        \item \textup{(Appropriate output domain).} For every element $v$ in \texttt{input\_domain}, $\function(v)$ is in \texttt{output\_domain}.
+        
+        \item \textup{(Domain-metric compatibility).} The domain \texttt{input\_domain} matches one of the possible domains listed in the definition of \texttt{input\_metric}, and likewise \texttt{output\_domain} matches one of the possible domains listed in the definition of \texttt{output\_metric}.
+        
+        \item \textup{(Stability guarantee).} For every pair of elements \texttt{v}, \texttt{w} in \texttt{input\_domain} and for every pair $(\din, \dout)$,  where $\din$ has the associated type for \texttt{input\_metric} and $\dout$ has the associated type for \texttt{output\_metric}, if $v,w$ are $\din$-close under \texttt{input\_metric} and $\texttt{relation}(\din, \dout) = \True$, then \texttt{function(v)}, \texttt{function(w)} are $\dout$-close under \texttt{output\_metric}.
```

Comment:
second occurrence of v,w missing texttt font

Reply:
Thanks -- fixed!